### DISCUSSION TITLE
Sending mail with attachment (Brevo vs SMTP

### DISCUSSION THREAD

**Discussion by rutonyru** [adonisjs/core#4684] [General]
Hi,
After having sent emails with attachment (one) through SMTP, I switched to Brevo.
I have an error when I try to attach a file in the message : 
```
file:///app/build/node_modules/@adonisjs/mail/build/src/transports/brevo.js:88
adonis_app  |         content: attachment.content.toString("base64")
adonis_app  |                                     ^
adonis_app  | 
adonis_app  | TypeError: Cannot read properties of undefined (reading 'toString')
adonis_app  |     at file:///app/build/node_modules/@adonisjs/mail/build/src/transports/brevo.js:88:37
adonis_app  |     at Array.map (<anonymous>)
adonis_app  |     at #preparePayload (file:///app/build/node_modules/@adonisjs/mail/build/src/transports/brevo.js:86:50)
adonis_app  |     at NodeMailerTransport2.send (file:///app/build/node_modules/@adonisjs/mail/build/src/transports/brevo.js:105:41)
adonis_app  |     at /app/build/node_modules/nodemailer/lib/mailer/index.js:218:34
adonis_app  |     at Mail._processPlugins (/app/build/node_modules/nodemailer/lib/mailer/index.js:266:20)
adonis_app  |     at /app/build/node_modules/nodemailer/lib/mailer/index.js:188:18
adonis_app  |     at processPlugins (/app/build/node_modules/nodemailer/lib/mailer/index.js:279:28)
adonis_app  |     at /app/build/node_modules/nodemailer/lib/mailer/index.js:287:17
adonis_app  |     at Mail._convertDataImages (/app/build/node_modules/nodemailer/lib/mailer/index.js:391:20)
``` 
It is like the file is not correctly given to the library ? 
It works fine if I switch back to SMTP.
My code :    `.attach(app.makePath(env.get('TEST_SEND_FILE')))`
I tried to attachData with a stream on the file but then i have an error from brevo api.

What should I do ?
Thanks,
Rutonyru

**Comment by thetutlage** [comment]
Hey, can you please create a new project that can help us reproduce this issue?

**Comment by rutonyru** [comment]
Hi,
Thanks for your help. As I have done some tests, I have more informations : 
  - I can send email and attachment via SMTP to a maildev server without any problem
  - I can't send email and attachment via Brevo configuration (using Brevo api Key)
  - I can send email and attachment via SMTP to brevo server, but only if I change the way I get the attachment file. 
    => Before : `.attach(app.makePath(env.get('TEST_SEND_FILE')))` 
    => After : `.attach(app.makePath(fileToSend))`
For me, it is like the "context" is not the same in the "await mail.send((message) => ...)" and outside ?
Well, I can make a project to reproduce the issue but I have to finish my production deployment ;) 
Thanks for your help,
Antoine

**Reply by RomainLanz** [reply]
Could you please share a repository with the minimum amount of code needed to reproduce the issue? It would be greatly appreciated.

**Comment by rutonyru** [comment]
Yes, i Can, of course. I will do After my holydays ( 😎),.. After the 18th
of August.
